.travis.yml

```diff
@@ -25,6 +25,6 @@ install:
   - pip install -e .
 
 script:
-  - ipcluster start -n 2 --daemon
+  - if [[ "$TRAVIS_OS_NAME" == "linux" ]]; then ipcluster start -n 2 --daemon ; fi
```

Comment:
Stopping the cluster is definitely good practice, but with Travis the whole virtual machine is destroyed after tests, so basically it would be just one more thing that can fail... :D

I'm not sure what you mean with a client since we already use one (sorry, no time to go through your code now).